### PR TITLE
update templates for diag_table to be consistent with model_configure

### PR DESCRIPTION
**Pull request by aerorahul**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.

- [ ] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories.  There is no change to the sub-components associated with this PR.

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model.  See issue #644 

## Description

The `diag_table` templates (and other e.g. MOM6, WW3, ICE) employ different templating than the ones used in `input.nml`, `model_configure` and `nems.configure`.  These files use the template defined via `@[VARIABLE]` which is replaced by `$VARIABLE` defined upstream.

The workflow team would like to use the templates from the UFS-weather-model for constructing the necessary input files.
This will allow the workflow and the model to use the same templates, albeit filled with application specific values for the templated variables.

No change of answers is expected from this PR.
There is no impact on the general user running regression tests.
There are no library or modulefiles updated or need changes with this PR.

### Issue(s) addressed

- Part of #644 

## Testing

These have not been tested by running the model on any platform in particular with the exception that the change produces the desired output `diag_file` from the template as before.

- [ ] hera.intel
- [ ] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel 
- [ ] cheyenne.gnu
- [ ] gaea.intel 
- [ ] jet.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3
- [ ] CI

## Dependencies

None.

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
@aerorahul Just FYI, the "templating" files were initially introduced when S2S was being run using the Python based CompsetRunner that Sam built. They are overdue for updating to whatever works best now that all applications are using the rt.sh infrastructure.

File: tests/edit_inputs.sh [hunk 1]
```diff
@@ -52,11 +52,12 @@ function edit_data_table {
...
-      -e "s/SYEAR/$SYEAR/g" \
-      -e "s/SMONTH/$SMONTH/g" \
-      -e "s/SDAY/$SDAY/g"
+  sed -e "s/@[YMD]/$SYEAR$SMONTH$SDAY/g" \
+      -e "s/@[ATMRES]/$ATMRES/g" \
+      -e "s/@[SYEAR]/$SYEAR/g" \
+      -e "s/@[SMONTH]/$SMONTH/g" \
+      -e "s/@[SDAY]/$SDAY/g" \
+      -e "s/@[SHOUR]/$SHOUR/g"
 }
```
Comment (DusanJovic-NOAA) [inline tests/edit_inputs.sh]: This function is not necessary anymore. We should use atparse.

Comment (aerorahul) [reply tests/edit_inputs.sh]: Would that mean replace:

```
edit_diag_table < ${PATHRT}/parm/${DIAG_TABLE:-diag_table_template} > diag_table
```
wtih
```
atparse < ${PATHRT}/parm/${DIAG_TABLE:-diag_table_template} > diag_table
```
on [L123](https://github.com/aerorahul/ufs-weather-model/blob/1e48750fb1d853c09c74616c1bfad086dea63e6e/tests/run_test.sh#L123) of `run_test.sh`?

Comment (junwang-noaa) [reply tests/edit_inputs.sh]: That is correct. We are going to replace all the edit_ functions with atparse. Would you please make this change, then we can commit it? Thanks

**Comment by aerorahul** [comment]
Github PR shows 29 files changed, when there are only 5 actually changed.
Closing and reopening this PR.

**Pull request closed without merging**